src/main/io/osd.c

```diff
@@ -1636,6 +1639,11 @@ static void osdUpdateStats(void)
         }
     }
 #endif
+#ifdef USE_GYRO_DATA_ANALYSE
+    if (featureIsEnabled(FEATURE_DYNAMIC_FILTER)) {
+        stats.max_fft = MAX(stats.max_fft, getMaxFFT());
+    }
+#endif
```

Comment:
Since you're accumulating the max value inside the dynamic filter there's no need to do it again here. You don't really need a stats value in the OSD code. Just retrieve the max from the dynamic filter when displaying the stats.